src/main/java/at/hannibal2/skyhanni/features/inventory/wardrobe/CustomWardrobe.kt

```diff
@@ -0,0 +1,570 @@
+import net.minecraftforge.fml.common.eventhandler.SubscribeEvent
+import java.awt.Color
+import kotlin.math.min
+import kotlin.time.Duration.Companion.milliseconds
+
+@SkyHanniModule
+object CustomWardrobe {
+
+    private val config get() = SkyHanniMod.feature.inventory.customWardrobe
+
+    private var displayRenderable: Renderable? = null
+    private var inventoryButton: Renderable? = null
+    private var editMode = false
+    private var waitingForInventoryUpdate = false
+
+    private var activeScale: Int = 100
+    private var currentMaxSize: Pair<Int, Int>? = null
+    private var lastScreenSize: Pair<Int, Int>? = null
+    private var guiName = "Custom Wardrobe"
+
+    @SubscribeEvent
+    fun onGuiRender(event: GuiContainerEvent.BeforeDraw) {
+        if (!isEnabled() || editMode) return
+        val renderable = displayRenderable ?: run {
+            update()
+            displayRenderable ?: return
+        }
+
+        val gui = event.gui
+        val screenSize = gui.width to gui.height
+
+        if (screenSize != lastScreenSize) {
+            lastScreenSize = screenSize
+            val didUpdate = updateScreenSize(screenSize)
+            if (didUpdate) return
+        }
+
+        val (width, height) = renderable.width to renderable.height
+        val pos = Position((gui.width - width) / 2, (gui.height - height) / 2)
+        if (waitingForInventoryUpdate && config.loadingText) {
+            val loadingRenderable = Renderable.string(
+                "§cLoading...",
+                scale = activeScale / 100.0
+            )
+            val loadingPos =
+                Position(pos.rawX + (width - loadingRenderable.width) / 2, pos.rawY - loadingRenderable.height)
+            loadingPos.renderRenderable(loadingRenderable, posLabel = guiName, addToGuiManager = false)
+        }
+
+        GlStateManager.translate(0f, 0f, 100f)
+        pos.renderRenderable(renderable, posLabel = guiName, addToGuiManager = false)
+        GlStateManager.translate(0f, 0f, -100f)
+        event.cancel()
+    }
+
+    // Edit button in normal wardrobe while in edit mode
+    @SubscribeEvent
+    fun onRenderOverlay(event: GuiRenderEvent.ChestGuiOverlayRenderEvent) {
+        if (!isEnabled()) return
+        if (!editMode) return
+        val gui = Minecraft.getMinecraft().currentScreen as? GuiContainer ?: return
+        val renderable = inventoryButton ?: addReEnableButton().also { inventoryButton = it }
+        val accessorGui = gui as AccessorGuiContainer
+        val posX = accessorGui.guiLeft + (1.05 * accessorGui.width).toInt()
+        val posY = accessorGui.guiTop + (accessorGui.height - renderable.height) / 2
+        Position(posX, posY).renderRenderable(renderable, posLabel = guiName, addToGuiManager = false)
+    }
+
+    @SubscribeEvent
+    fun onInventoryClose(event: InventoryCloseEvent) {
+        waitingForInventoryUpdate = false
+        DelayedRun.runDelayed(500.milliseconds) {
```

Comment:
only do this if the feature is enabled & in wardrobe. no need to create a run delayed task each inventory close, even when in another inventory